### PR TITLE
Ns/0.6/core crypto versionize

### PR DESCRIPTION
**Pull request by nsarlin-zama**
closes: _please link all relevant issues_

### PR content/description

- add versioning for missing core_crypto_entities

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/core_crypto/entities/gsw_ciphertext.rs [hunk 1]
```diff
@@ -153,7 +156,8 @@ use crate::core_crypto::commons::traits::*;
 /// Simply use the
 /// [`LWE decryption algorithm`](`crate::core_crypto::algorithms::decrypt_lwe_ciphertext`)
 /// on one of the LWE ciphertexts contained in the Lev ciphertext.
-#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize, Versionize)]
+#[versionize(GswCiphertextVersions)]
 pub struct GswCiphertext<C: Container> {
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/entities/gsw_ciphertext.rs:161]: we can skip this one for now, we don't have constructors nor code associated

File: tfhe/src/core_crypto/backward_compatibility/entities/gsw_ciphertext.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use tfhe_versionable::VersionsDispatch;
+
+use crate::core_crypto::prelude::{Container, GswCiphertext};
+
+#[derive(VersionsDispatch)]
+pub enum GswCiphertextVersions<C: Container> {
+    V0(GswCiphertext<C>),
+}
```
Comment (IceTDrinker) [inline tfhe/src/core_crypto/backward_compatibility/entities/gsw_ciphertext.rs:8]: we can skip this one for now, it cannot be built by users

File: tfhe/src/core_crypto/entities/gsw_ciphertext.rs (hunk 1 above)
Comment (nsarlin-zama) [reply tfhe/src/core_crypto/entities/gsw_ciphertext.rs]: done

File: tfhe/src/core_crypto/backward_compatibility/entities/gsw_ciphertext.rs (hunk 2 above)
Comment (nsarlin-zama) [reply tfhe/src/core_crypto/backward_compatibility/entities/gsw_ciphertext.rs]: done

**Review by IceTDrinker — APPROVED** [review]
Big fan of that proc macro

**Comment by IceTDrinker** [comment]
same thing as the other PR

**Review by IceTDrinker — APPROVED** [review]
thanks !

**Pull request merged**